services/queue/src/artifacts.js

```diff
@@ -33,47 +121,6 @@ builder.declare({
     'expiration point. This features makes it feasible to upload large',
     'intermediate artifacts from data processing applications, as the',
     'artifacts can be set to expire a few days later.',
-    '',
-    'We currently support "S3 Artifacts" for data storage.',
-    '',
-    '**S3 artifacts**, is useful for static files which will be',
-    'stored on S3. When creating an S3 artifact the queue will return a',
-    'pre-signed URL to which you can do a `PUT` request to upload your',
-    'artifact. Note that `PUT` request **must** specify the `content-length`',
-    'header and **must** give the `content-type` header the same value as in',
-    'the request to `createArtifact`.',
-    '',
-    '**Redirect artifacts**, will redirect the caller to URL when fetched',
-    'with a a 303 (See Other) response.  Clients will not apply any kind of',
-    'authentication to that URL.',
-    '',
-    '**Link artifacts**, will be treated as if the caller requested the linked',
-    'artifact on the same task.  Links may be chained, but cycles are forbidden.',
-    'The caller must have scopes for the linked artifact, or a 403 response will',
-    'be returned.',
-    '',
-    '**Error artifacts**, only consists of meta-data which the queue will',
-    'store for you. These artifacts are only meant to indicate that you the',
-    'worker or the task failed to generate a specific artifact, that you',
-    'would otherwise have uploaded. For example docker-worker will upload an',
-    'error artifact, if the file it was supposed to upload doesn\'t exists or',
-    'turns out to be a directory. Clients requesting an error artifact will',
-    'get a `424` (Failed Dependency) response. This is mainly designed to',
-    'ensure that dependent tasks can distinguish between artifacts that were',
-    'suppose to be generated and artifacts for which the name is misspelled.',
-    '',
-    '**Artifact immutability**, generally speaking you cannot overwrite an',
-    'artifact when created. But if you repeat the request with the same',
-    'properties the request will succeed as the operation is idempotent.',
-    'This is useful if you need to refresh a signed URL while uploading.',
-    'Do not abuse this to overwrite artifacts created by another entity!',
-    'Such as worker-host overwriting artifact created by worker-code.',
-    '',
-    '**Immutability Special Cases**:',
-    '',
-    '* A `reference` artifact can replace an existing `reference` artifact`.',
-    '* A `link` artifact can replace an existing `reference` artifact`.',
-    '* Any artifact\'s `expires` can be extended.',
```

Comment:
It this removed here because it is moved to ui/docs/reference/platform/queue/artifacts.mdx ?

I liked having it in the reference section, because then it is generated e.g. in the go client. With it no longer there, users of the go client won't discover this summary, I wonder if it might be better to keep it here?

Reply:
Yes, because the artifact types are relevant to more than just createArtifact.  For example, artifact users should know that reference artifacts cannot be cached.

It's still in the reference section, just not tied to one specific endpoint.  Go (and rust and Python) client users couldn't understand Taskcluster as a whole just by reading the go docs, anyway -- they also need this kind of more-than-an-API-method context.

Reply:
I think it would be best to put it in the APIBuilder description of the Queue service. This way, it will still be included in the package docs of the queue clients (e.g. [here](https://pkg.go.dev/github.com/taskcluster/taskcluster/v42@v42.1.1/clients/client-go/tcqueue#pkg-overview), but won't be bound to specific API endpoints. My concern with moving service docs outside of the schemas and references is that information about the services doesn't have a single home, and it isn't clear where it should go. I think if we keep it in the service description, the clients should get it, the docs site, and anything that crawls the APIs etc. It is also much closer to the code, and having docs directly with the code seems like a good practice.

Is there an advantage of having it in ui/docs/reference/platform/queue/artifacts.mdx instead of services/queue/src/api.js ?

Reply:
Fair enough.  The advantage is that it shows up in the table of contents, but i can add some links.